lib/Twig/Loader/Filesystem.php

```diff
@@ -23,13 +23,17 @@ class Twig_Loader_Filesystem implements Twig_LoaderInterface, Twig_ExistsLoaderI
     protected $cache = array();
     protected $errorCache = array();
 
+    private $rootPath;
+
     /**
      * Constructor.
      *
-     * @param string|array $paths A path or an array of paths where to look for templates
+     * @param string|array $paths    A path or an array of paths where to look for templates
+     * @param string       $rootPath The root path common to all relative paths
      */
-    public function __construct($paths = array())
+    public function __construct($paths = array(), $rootPath = null)
     {
+        $this->rootPath = realpath(null !== $rootPath ? $rootPath : getcwd()).DIRECTORY_SEPARATOR;
```

Comment:
I would say that we don't want to support this use case (if everything is in a phar, then using an absolute path is not really an issue anymore). So, if `realpath` returns false here, I would prefer to throw an exception.